benchmark/profile_throughput.py

```diff
@@ -190,7 +190,7 @@ async def _inference(self, req_queue: Queue, session_id: int, temperature: float
                 await generator.aclose()
 
             # for pytorch engine to restart a session
-            if isinstance(model_inst, EngineInstance):
+            if hasattr(model_inst, '_is_pytorch_engine'):
```

Comment:
How about adding `self.backend` in the constructor to indicate what engine it is?